### DISCUSSION TITLE
what is the meaning about fine-tuning the backbone

### DISCUSSION THREAD

**Discussion by Bo396543018** [fredzzhang/spatio-attentive-graphs#56] [Q&A] [answered]
Hi, Fred. I had a question for the last sentence in your paper.

> we observe a further 1 mAP improvement from fine-tuning the backbone.

Did it mean to use hicodet finetuned detector to train SCG model? The default setting is loading coco pretrained detector,  I have not found any code related this option.

**Comment by fredzzhang** [comment, marked as answer]
Hi, @Bo396543018 

The backbone refers to the feature backbone for the HOI model, not the detector. A ResNet50 is used to extract appearance features for each human and object instance. Fine-tuning this backbone re-calibrates the appearance features based on the HICO-DET dataset. You can find the related code in `main.py`

```python
    # Seperate backbone parameters from the rest
    param_group_1 = []
    param_group_2 = []
    for k, v in engine.fetch_state_key('net').named_parameters():
        if v.requires_grad:
            if k.startswith('module.backbone'):
                param_group_1.append(v)
            elif k.startswith('module.interaction_head'):
                param_group_2.append(v)
            else:
                raise KeyError(f"Unknown parameter name {k}")
    # Fine-tune backbone with lower learning rate
    optim = torch.optim.AdamW([
        {'params': param_group_1, 'lr': args.learning_rate * args.lr_decay},
        {'params': param_group_2}
        ], lr=args.learning_rate,
        weight_decay=args.weight_decay
    )
    lambda1 = lambda epoch: 1. if epoch < args.milestones[0] else args.lr_decay
    lambda2 = lambda epoch: 1. if epoch < args.milestones[0] else args.lr_decay
    lr_scheduler = torch.optim.lr_scheduler.LambdaLR(
        optim, lr_lambda=[lambda1, lambda2]
    )
```
Hope that answers your question.

Cheers,
Fred

**Reply by Bo396543018** [reply]
Hi,Fread
Thank you for your explanation, I can already understand the meaning of this sentence. 
And I have another question，instead of using coco pretrained backbone，Is it better use hicodet pretrained backbone to train HOI model? I'll do this experiment in next week.

**Reply by fredzzhang** [reply]
By pre-trained backbone, I assume you meant the backbone of the detector? I think it wouldn't make difference. The backbone will be fine-tuned anyways, so different pre-trained backbones shouldn't matter here.
